nylon-threads/src/test/java/com/palantir/nylon/threads/NylonExecutorTest.java

```diff
@@ -224,4 +237,87 @@ void testMaxQueueSize() throws InterruptedException {
                     .isTrue();
         }
     }
+
+    @ParameterizedTest
+    @ValueSource(ints = {1, 2, 3, 4, 5})
+    void testMaxThreadsLimitsConcurrentlyRunningThreads(int maxThreads) {
+        AtomicInteger threadsCreated = new AtomicInteger();
+        ExecutorService delegate = Executors.newCachedThreadPool(new ThreadFactoryBuilder()
+                .setDaemon(true)
+                .setNameFormat("test-%d")
+                .setThreadFactory(r -> {
+                    int id = threadsCreated.incrementAndGet();
+                    Thread thread = Executors.defaultThreadFactory().newThread(r);
+                    thread.setName("thread-" + id);
+                    return thread;
+                })
+                .build());
+        try {
+            CountDownLatch countDownLatch = new CountDownLatch(maxThreads);
+            int tasks = 3 * maxThreads;
+            List<ListenableFuture<String>> futures = new ArrayList<>(tasks);
+            List<Instant> threadStarts = Collections.synchronizedList(new ArrayList<>(maxThreads));
+            List<Instant> threadEnds = Collections.synchronizedList(new ArrayList<>(maxThreads));
+            ListeningExecutorService executor = MoreExecutors.listeningDecorator(NylonExecutor.builder()
+                    .name("foo")
+                    .executor(delegate)
+                    .maxThreads(maxThreads)
+                    .build());
+            for (int i = 0; i < tasks; i++) {
+                futures.add(executor.submit(() -> {
+                    threadStarts.add(Instant.now());
+                    countDownLatch.countDown();
+                    countDownLatch.await();
```

Comment:
`CyclicBarrier` feels more appropriate here

```suggestion
                    barrier.await();
```